src/gobapi/api.py

```diff
@@ -96,24 +102,33 @@ def _collection(catalog_name, collection_name):
     if get_collection(catalog_name, collection_name):
         page = int(request.args.get('page', 1))
         page_size = int(request.args.get('page_size', 100))
-        result, links = _entities(catalog_name, collection_name, page, page_size)
+        view = request.args.get('view', None)
+        result, links = _entities(catalog_name, collection_name, page, page_size, view)
         return hal_response(data=result, links=links), 200, {'Content-Type': 'application/json'}
     else:
         return not_found(f'{catalog_name}.{collection_name} not found')
 
 
-def _entity(catalog_name, collection_name, entity_id):
+def _entity(catalog_name, collection_name, entity_id, view=None):
     """Returns the entity within the specified collection with the specified id
 
     An individual entity is returned.
 
     :param catalog_name: e.g. meetbouten
     :param collection_name: e.g. meting
     :param entity_id: unique identifier of the entity
+    :param view: the database view that's being used to get the entity, defaults to the entity table
     :return:
     """
+    view = request.args.get('view', None)
+
     if get_collection(catalog_name, collection_name):
-        result = get_entity(collection_name, entity_id)
+        if view:
+            view_name = f"{catalog_name}_{collection_name}_{view}"
+        else:
+            view_name = None
+
+        result = get_entity(collection_name, entity_id, view_name)
         return (hal_response(result), 200, {'Content-Type': 'application/json'}) if result is not None else not_found(
             f'{catalog_name}.{collection_name}:{entity_id} not found')
     else:
```

Comment:
In general: try to limit the duplicated code in get_entities and get_entity. The both versions of each method are now practically the same and contain some unnecessary duplicated logic